src/main/java/com/facebook/openwifirrm/optimizers/tpc/MeasurementBasedApClientTPC.java

```diff
@@ -226,38 +230,55 @@ private int computeTxPowerForRadio(
 				newTxPower
 			);
 
-			// If this exceeds max tx power, repeat for (MCS - 1)
-			if (newTxPower > MAX_TX_POWER) {
+			// If this exceeds max possible tx power, repeat for (MCS - 1)
+			if (newTxPower > maxTxPower) {
 				logger.info(
 					"Device {}: computed tx power > maximum {}, trying with mcs - 1",
 					serialNumber,
-					MAX_TX_POWER
+					maxTxPower
 				);
 				if (--mcs >= 0) {
 					continue;
 				} else {
 					logger.info(
 						"Device {}: already at lowest MCS, setting to minimum tx power {}",
 						serialNumber,
-						MIN_TX_POWER
+						minTxPower
 					);
-					newTxPower = MIN_TX_POWER;
+					newTxPower = minTxPower;
 				}
 			}
 
 			// If this is below min tx power, set to min
-			else if (newTxPower < MIN_TX_POWER) {
+			else if (newTxPower < minTxPower) {
 				logger.info(
 					"Device {}: computed tx power < minimum {}, using minimum",
 					serialNumber,
-					MIN_TX_POWER
+					minTxPower
 				);
-				newTxPower = MIN_TX_POWER;
+				newTxPower = minTxPower;
 			}
 
 			break;
 		} while (true);
 
+		int txPowerIndex = 0;
+		for (
+			txPowerIndex = 0;
+			txPowerIndex < txPowerChoices.size() &&
+				txPowerChoices.get(txPowerIndex) < newTxPower;
+			txPowerIndex++
+		) {
+
```

Comment:
did you mean to leave this block?

Reply:
Yes. Just want `txPowerIndex` to be the first index where the element is larger than or equal to `newTxPower`. No further computation is needed in this block.

Reply:
makes sense, but I guess as a personal thing I would expect a for loop to be just giving iteration logic and then the check would come inside the block with a `break` or something